pyquil/tests/conftest.py

```diff
@@ -189,3 +189,12 @@ def local_qvm_quilc():
 
     with local_qvm() as context:
         yield context
+
+
+def pytest_addoption(parser):
+    parser.addoption("--use_seed", action="store", default=True, help="run some tests faster using random seed")
```

Comment:
1. Changed `use_seed` to `use-seed`.
2. Picked option iii, but made the str to bool conversion function private
3. I'd rather stick with `--use-seed` than go for something like `--test-carefully` because the latter could be quite misleading (see my earlier comments about how even the fixed seed tests are *careful* enough), and the former makes very explicit what is being done. I would even argue that we use this flag in other places across the test suite where such a distinction between seeded vs averaged tests needs to be made.

Additionally, I've changed the default value to `False` so that Semaphore still runs the longer, and more complete versions of the tests, and if someone needs to iterate on `pyquil.operator_estimation` quickly enough, they can use the `use-seed` option.